crates/sui-types/src/transaction.rs

```diff
@@ -2200,12 +2261,14 @@ impl InputObjects {
     }
 
     /// The version to set on objects created by the computation that `self` is input to.
-    /// Guaranteed to be strictly greater than the versions of all input objects.
-    pub fn lamport_timestamp(&self) -> SequenceNumber {
+    /// Guaranteed to be strictly greater than the versions of all input objects and objects
+    /// received in the transaction.
+    pub fn lamport_timestamp(&self, receiving_objects: &[ObjectRef]) -> SequenceNumber {
```

Comment:
I think you could just add receiving objects to the InputObjects data structure, which would simplify quite some code?

Reply:
I'm hesitant to do this, as it would simplify this, but also make other portions more complicated/confusing and break some of the APIs for `InputObjects` since not all object IDs in the input objects would actually have an `Object` for them at input time, and we wouldn't be able to accurately compute transaction dependencies from it (since we can only do that post-execution).

Reply:
It won't be part of the Vec<Object> in InputObjects, but a separate field, and if it's confusing you could also rename the InputObjects type. The point is that `InputObjects` could contain both tx input objects and receiving objects.

Reply:
By the way, this shouldn't be blocking for you. Feel free to add a TODO and address on the naming and typing and etc. latter.
One way to make things clearer is to name `InputObjects` as `TransactionObjects`, and then rename the input objects that we refer to today as authenticated input objects, to distinguish from receiving objects.